File: BE/order-service/src/main/java/com/vietnl/orderservice/domain/models/entities/OrderItem.java
package com.vietnl.orderservice.domain.models.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UuidGenerator;

import java.math.BigDecimal;
import java.util.UUID;

@Entity
@Table(name = "order_items")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class OrderItem {

    @Id
    @GeneratedValue
    @UuidGenerator
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    // Tham chiếu đến catalog-service menu item
    @Column(name = "menu_item_id", nullable = false)
    private String menuItemId;

    @Column(name = "food_name", nullable = false)
    private String foodName;

    @Column(name = "unit_price", nullable = false, precision = 18, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "quantity", nullable = false)
    private int quantity = 1;

    @Column(name = "note", length = 300)
    private String note;

    // Trạng thái chế biến: PENDING, COOKING, READY, SERVED
    @Column(name = "kitchen_status")
    private String kitchenStatus = "PENDING";

    public BigDecimal getSubtotal() {
        return unitPrice.multiply(BigDecimal.valueOf(quantity));
    }
}

File: FE/mobile/app/src/main/java/com/food/order/data/model/OrderItem.kt
package com.food.order.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class OrderItem(
    @SerializedName("menuItemId") val foodId: String? = null,
    @SerializedName("foodName") val foodName: String? = null,
    @SerializedName("foodImage") val foodImage: String? = null,
    @SerializedName("unitPrice") val price: Double? = null,
    @SerializedName("quantity") val quantity: Int? = null,
    @SerializedName("note") val note: String? = null,
    @SerializedName("unit") val unit: String? = null,
    @SerializedName("id") val id: String? = null,
    @SerializedName("kitchenStatus") val kitchenStatus: String? = null
) : Serializable
